sklearn/neural_network/random_basis_function.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+
+from .base import ACTIVATIONS
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import check_random_state
+from ..utils.extmath import safe_sparse_dot
+from ..utils.validation import check_array
+
+
+class RandomBasisFunction(BaseEstimator, TransformerMixin):
+
+    """Random basis activation.
+
+    The algorithm uses the number of features of the input data to randomly 
+    generate coefficient parameters based on the uniform probability 
+    distribution.
+
+    Using these coefficient parameters, the algorithm can transform 
+    the data into a different dimensional space.
+
+    Parameters
+    ----------
+    intercept : boolean, default True
+        Whether to randomly generate an intercept. 
+
+    weight_scale : float, default 'auto'
+        Initializes and scales the input-to-hidden weights.
+        The weight values will range between plus and minus
```

Comment:
Should this be "if 'auto'" and state, otherwise range from -weight_scale to +weight_scale?